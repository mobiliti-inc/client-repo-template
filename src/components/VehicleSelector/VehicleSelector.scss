@import "../../../styles/_variables.scss";

.vehicle-selector {
    display: flex;
    flex-direction: column;
    overflow-y: visible;
    justify-content: center;
    align-items: center;
    align-content: center;

    opacity: 0.43;
    cursor: pointer;
    margin-right: 10px;
    &--active {
        opacity: 1;
    }

    &__image {
        height: 190px;	
        width: 331px;
        margin-bottom: 45px;
        & > img {
            height: 100%;	
            width:100%;
        }
    }

    &__name {
        display: flex;
        height: 21px;	
        color:  $darker-blue;
        font-family: $hanken-sans;
        font-size: 19px;	
        font-weight: 500;	
        letter-spacing: -0.42px;	
        line-height: 21px;	
        text-align: center;
        margin-bottom: 25px;
    }

    &__selected-indicator {
        height: 4px;	
        width: 275px;
        background-color:  transparent;
        &--active {
            background-color:  $blue;
        }
    }
}