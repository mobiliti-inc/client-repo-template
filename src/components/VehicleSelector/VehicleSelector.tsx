import styles from './VehicleSelector.scss';

type VehicleSelectorProps = {
	vehicleName: string,
	vehicleImageUrl: string,
	isActive?: boolean,
	onVehicleSelected: (...args: any[]) => any
};

const VehicleSelector: React.SFC<VehicleSelectorProps> = ({
	vehicleName,
	vehicleImageUrl,
	isActive,
	onVehicleSelected
}) => (
		<div
			role="presentation"
			onClick={onVehicleSelected}
			styleName={`vehicle-selector ${isActive ? 'vehicle-selector--active' : ''}`}
		>
			<div styleName="vehicle-selector__image">
				<img src={vehicleImageUrl} alt="vehicle-selector" />
			</div>
			<div styleName="vehicle-selector__name">{vehicleName}</div>

			<span
				styleName={`vehicle-selector__selected-indicator ${isActive ? 'vehicle-selector__selected-indicator--active' : ''}`}
			>
				&nbsp;
		</span>
		</div>
	);

// VehicleSelector.defaultProps = {
// 	isActive: false
// };

export default CSSModules(VehicleSelector, styles, { allowMultiple: true });
