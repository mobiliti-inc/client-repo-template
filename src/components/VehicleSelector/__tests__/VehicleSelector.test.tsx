import VehicleSelector from '../VehicleSelector';

describe('VehicleSelector', () => {
	const defaultProps = {
		vehicleName: 'Vehicle name',
		vehicleImageUrl: 'Vehicle image url',
		isActive: false,
		onVehicleSelected: jest.fn(),
	};
	it('renders with isActive set to false', () => {
		const subject = shallow(<VehicleSelector
			{...defaultProps}
		/>);
		matchSnapshot(subject);
	});

	it('renders with isActive set to true', () => {
		const subject = shallow(<VehicleSelector
			{...defaultProps}
			isActive
		/>);
		matchSnapshot(subject);
	});
});
