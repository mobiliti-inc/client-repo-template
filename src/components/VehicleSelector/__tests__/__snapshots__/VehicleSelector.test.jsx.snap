// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleSelector renders with isActive set to false 1`] = `
<div
  className="vehicle-selector"
  onClick={[MockFunction]}
  role="presentation"
>
  <div
    className="vehicle-selector__image"
  >
    <img
      alt="vehicle-selector"
      src="Vehicle image url"
    />
  </div>
  <div
    className="vehicle-selector__name"
  >
    Vehicle name
  </div>
  <span
    className="vehicle-selector__selected-indicator"
  >
     
  </span>
</div>
`;

exports[`VehicleSelector renders with isActive set to true 1`] = `
<div
  className="vehicle-selector vehicle-selector--active"
  onClick={[MockFunction]}
  role="presentation"
>
  <div
    className="vehicle-selector__image"
  >
    <img
      alt="vehicle-selector"
      src="Vehicle image url"
    />
  </div>
  <div
    className="vehicle-selector__name"
  >
    Vehicle name
  </div>
  <span
    className="vehicle-selector__selected-indicator vehicle-selector__selected-indicator--active"
  >
     
  </span>
</div>
`;
